.github/BUILD.md

```diff
@@ -13,81 +13,109 @@ If the answer is yes, please read on. Otherwise, make sure to head on over [to t
 
 ```
-cd src\Umbraco.Web.UI.Login
-npm i
-npm run dev
+REM npm ci --no-fund --no-audit --prefer-offline
+REM npm run build:for:cms
 ```
 
-If you just want to build the Login screen to `Umbraco.Web.UI` then instead of running `dev`, you can do: `npm run build`.
+Just be careful not to include this change in your PR.
+
+### Front-end only changes
+
+Conversely, if you are working on front-end only, you want to build the back-end once and then run it. Before you do so, update the configuration in `appSettings.json` to add the following under `Umbraco:Cms:Security`:
+
+```
+"BackOfficeHost": "http://localhost:5173",
+"AuthorizeCallbackPathName": "/oauth_complete",
+"AuthorizeCallbackLogoutPathName": "/logout",
+"AuthorizeCallbackErrorPathName": "/error"
+```
 
-To run the C# portion of the project, either hit <kbd>F5</kbd> to begin debugging, or manually using the command line:
+Then run Umbraco from the command line.
 
 ```
-dotnet watch --project .\src\Umbraco.Web.UI\Umbraco.Web.UI.csproj
+cd <solution root>\src\Umbraco.Web.UI
+dotnet run --no-build
 ```
 
-**The initial C# build might take a _really_ long time (seriously, go and make a cup of coffee!) - but don't worry, this will be faster on subsequent runs.**
+In another terminal window, run the following to watch the front-end changes and launch Umbraco using the URL indicated from this task.
 
-When the page eventually loads in your web browser, you can follow the installer to set up a database for debugging. You may also wish to install a [starter kit][starter kits] to ease your debugging.
+```
+cd <solution root>\src\Umbraco.Web.UI.Client
+npm run dev:server
+```
 
-#### Debugging with Visual Studio
+You'll find as you make changes to the front-end files, the updates will be picked up and your browser refreshed automatically.
 
-In order to build the Umbraco source code locally with Visual Studio, first make sure you have the following installed.
+> [!NOTE]
+> The caching for the back office has been described as 'aggressive' so we often find it's best when making back office changes to [disable caching in the browser (check "Disable cache" on the "Network" tab of developer tools)][disable browser caching] to help you to see the changes you're making.
 
--   [Visual Studio 2022 v17+ with .NET 7+](https://visualstudio.microsoft.com/vs/) ([the community edition is free](https://www.visualstudio.com/thank-you-downloading-visual-studio/?sku=Community&rel=15) for you to use to contribute to Open Source projects)
--   [Node.js v20+](https://nodejs.org/en/download/)
--   npm v10+ (installed with Node.js)
--   [Git command line](https://git-scm.com/download/)
+Whilst most of the backoffice code lives in `Umbraco.Web.UI.Client`, the login screen is in a separate project. If you do any work with that you can build with:
 
-The easiest way to get started is to open `umbraco.sln` in Visual Studio.
+```
+cd <solution root>\src\Umbraco.Web.UI.Login
+npm run build
```

Comment:
do an `npm ci` first if it's never had the package restore process run before?

Reply:
I've added a single reference to when you might have to do this and how you should do it.